idaes_examples/notebooks/docs/unit_models/operations/tests/test_eg_h2o_ideal.py

```diff
@@ -0,0 +1,398 @@
+
+    @pytest.mark.skipif(solver is None, reason="Solver not available")
+    @pytest.mark.component
+    def test_initialize(self, model):
+        orig_fixed_vars = fixed_variables_set(model)
+        orig_act_consts = activated_constraints_set(model)
+
+        model.props.initialize(optarg={"tol": 1e-6})
+
+        assert degrees_of_freedom(model) == 0
+
+        fin_fixed_vars = fixed_variables_set(model)
+        fin_act_consts = activated_constraints_set(model)
+
+        assert len(fin_act_consts) == len(orig_act_consts)
+        assert len(fin_fixed_vars) == len(orig_fixed_vars)
+
+        for c in fin_act_consts:
+            assert c in orig_act_consts
+        for v in fin_fixed_vars:
+            assert v in orig_fixed_vars
+
+    @pytest.mark.skipif(solver is None, reason="Solver not available")
+    @pytest.mark.component
+    def test_solve(self, model):
+        results = solver.solve(model)
+
+        # Check for optimal solution
+        assert check_optimal_termination(results)
+
+    @pytest.mark.skipif(solver is None, reason="Solver not available")
+    @pytest.mark.component
+    def test_solution(self, model):
+        # Check results
+        assert model.props[1].flow_mol_phase_comp[
+            "Liq", "water"
+        ].value == pytest.approx(100, abs=1e-4)
+        assert model.props[1].flow_mol_phase_comp[
+            "Liq", "ethylene_glycol"
+        ].value == pytest.approx(100, abs=1e-4)
+
+        assert value(
+            model.props[1].temperature
+        ) == pytest.approx(300, abs=1e-4)
+        assert value(
+            model.props[1].pressure
+        ) == pytest.approx(101325, abs=1e-4)
+
+
+class TestPerrysProperties(object):
+    @pytest.fixture(scope="class")
+    def model(self):
+        model = ConcreteModel()
+        model.params = GenericParameterBlock(**config_dict)
+
+        model.props = model.params.build_state_block([1], defined_state=True)
+
+        model.props[1].calculate_scaling_factors()
+
+        return model
+
+    @pytest.fixture(scope="class")
+    def density_temperatures(self):
+        # water, ethylene glycol reference temperatures
+        # from Perry's Chemical Engineers' Handbook 7th Ed. 2-94 to 2-98
+        components = ["water", "ethylene_glycol"]
+        temperatures = dict(
+            zip(
+                components,
+                [[273.16, 333.15], [260.15, 719.7]]
+                )
+            )
+
+        return temperatures
+
+    @pytest.fixture(scope="class")
+    def densities(self):
+        # water, ethylene glycol densities from
+        # Perry's Chemical Engineers' Handbook 7th Ed. 2-94 to 2-98
+        components = ["water", "ethylene_glycol"]
+        densities = dict(
+            zip(
+                components,
+                [[55.583, 54.703], [18.31, 5.234]]
+                )
+            )
+
+        return densities
+
+    @pytest.fixture(scope="class")
```

Comment:
Why are these `fixtures` and not just module/class level dicts?

Reply:
(sorry for the "lecturing" tone, I just wanted to provide some motivation for my remarks)